network/address.go

```diff
@@ -0,0 +1,156 @@
+package network
+
+import (
+	"net"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+// ConnType is a type to represent what is the type of a Conn
+// It can be of different types such as TCP, TLS etc.
+type ConnType string
+
+// Address contains the ConnType and the actual network address. It is used to connect
+// to a remote host with a Conn and to listen by a Listener.
+// A network address is comprised of the IP address and the port number joined
+// by a colon.
+// For the moment, there is no IPv6 support.
+type Address string
+
+const (
+	// PlainTCP represents a vanilla TCP connection
+	PlainTCP ConnType = "tcp"
+	// TLS represents a TLS encrypted connection over TCP
+	TLS = "tls"
+	// PURB represents a PURB encryption connection over TCP
+	PURB = "purb"
+	// Local represents a channel based connection type
+	Local = "local"
+	// InvalidConnType represents a non valid connection type
+	InvalidConnType = "wrong"
+)
+
+// typeAddressSep is the separator between the type of connection and the actual
+// ip address.
+const typeAddressSep = "://"
+
+func connType(t string) ConnType {
```

Comment:
I'm not sure. It depends on our uses cases. If we go for returning an error, then on every operation of the `Address` we have to return an error also (because if ConnType is not valid we should not return anything OK in every other operations). It makes using this whole file a lot more cumbersome than it has to be to my feeling.
That's why there is a `Valid()` method for. If the address is `Valid()` (when we read it from file / user input / whatever) then we can relax and call all the methods without checking each time if something's wrong while we know the address is correct after the first time we checked it. It's not a new operation each time, it's operating with the same data. If there was some operation generating some new data each time (like read from a randomness source) then YES, ok we have to make sure, but here it's not the case.

Printing, or switching over different connections type
1) Using an `InvalidConnType`:
```
if  !a.Valid() {
    fmt.Println("aie something's wrong")
   os.Exit(1)
}
fmt.Printf("Address Conntype = %s",a.ConnType())
... // other method
switch a.ConnType() {
    case PlainTCP:
         // do TCP
    case Local:
         // do Local
    case InvalidConnType:
         // do something !
}
```
2) Using errors:
```
if  !a.Valid() {
    fmt.Println("aie something's wrong")
   os.Exit(1)
}
ct,err := a.ConnType()
if err != nil {
   fmt.Println("Something's stil wrong??")
}
fmt.Printf("Address ConnType: %s", ct)
.... // other method
ct,err := a.ConnType() 
if err != nil {
fmt.Printf("Aie still something wrong")
}
switch ct { 
    case PlainTCP:
         // do TCP
    case Local:
         // do Local
    case InvalidConnType:
         // do something !
}
```